conf/machine/raspberrypi3.conf

```diff
@@ -7,10 +7,10 @@ require conf/machine/include/tune-cortexa7.inc
 include conf/machine/include/rpi-base.inc
 
 MACHINE_EXTRA_RRECOMMENDS += "\
-    linux-firmware-bcm43430 \
-    linux-firmware-bcm43455 \
-    linux-firmware-bcm43430a1-hcd \
-    linux-firmware-bcm4345c0-hcd \
+    firmware-nonfree-rpi-distro-bcm43430 \
+    firmware-nonfree-rpi-distro-bcm43455 \
+    bluez-firmware-rpi-distro-bcm43430a1-hcd \
+    bluez-firmware-rpi-distro-bcm4345c0-hcd \
```

Comment:
@kraj Are you asking for two changes: `firmware-nonfree-` to `linux-firmware-nonfree-` and `-rpi-distro` to `-rpidistro`?  Or do you want both `bluez-firmware` and `firmware-nonfree` to combine as `linux-firmware-nonfree`?

tl;dr I'd rather keep both base names as-is.  I'm not tightly committed to `-rpi-distro`; I picked it as closer to `RPi-Distro` than `-rpidistro` would be, but we could smash that together.  (I also see some precedent for a prefix `rpidistro-firmware-nonfree` given Ubuntu's `dahdi-firmware-nonfree`, but I don't want to go there.)

As background, on raspbian the package names are `firmware-brcm80211` (containing all of what they get from `firmware-nonfree`), and `bluez-firmware`.

Combining `bluez-firmware` and `firmware-nonfree` is not workable: they are two distinct upstream repositories, and making them look like one (and part of `linux-firmware`) is the mess in #298 that I'm trying to fix.  They're both unversioned, and if we go that route we can't encode the SHA1 in the package name as a clue to what it contains, which in the past has been the only clue why somebody's system doesn't work (outdated firmware).  They have to be in different recipes, and should not (in my opinion) make themselves look like `linux-firmware`.

Renaming `firmware-nonfree` to `linux-firmware-nonfree` is possible, but it doesn't reflect the origin repository name, doesn't match Debian (which calls it `firmware-linux-nonfree`) and I believe incorrectly suggests a relationship to the other `linux-firmware-SOCID` packages that does not exist.  [`firmware-nonfree`](https://github.com/RPi-Distro/firmware-nonfree) appears to be a fork of Debian's internal archive of restricted firmware, with a snapshot of `linux-firmware` from an unidentified tar file dumped on top of it followed by a few modified files.  It's not `linux-firmware`, nor is `bluez-firmware` what's in the same package in Debian generically.

I feel the least confusing path is the names I've proposed, or with `-rpidistro` instead of `-rpi-distro`.

Reply:
@pabigot First of all thanks for picking this up. I'm not keen on the naming as well. Mainly for the fact that we removed "linux". I'd at least keep `linux-firmware` as a prefix. Now about nonfree and rpidistro, I'd as well change it in a way to keep only one. These two qualifiers have the same purpose of making this recipe distinct from the main upstream `linux-firmware`. My vote would be for `linux-firmware-rpidistro-ABC`. It shows that:

- it is a linux-firmware fork
- the fork is from RPi Distro
- it's not insanely long

Reply:
@agherzan I don't understand the strong desire to keep "linux" in the name.  The content of these packages is vendor blobs loaded onto the SoC to provide functionality that isn't linux-specific, comes from sources that don't name it that way, and only one of the four blobs even exists in linux-firmware as an obsolete version.

Nonetheless, it's your bikeshed.

To be absolutely clear so I don't have to rework this multiple times, what you want is:
```
bluez-firmware-rpi-distro-bcm43430a1-hcd-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
bluez-firmware-rpi-distro-bcm4345c0-hcd-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
bluez-firmware-rpi-distro-cypress-license-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
firmware-nonfree-rpi-distro-bcm43430-0.0+git0+b518de45ce-r0.1.noarch.rpm
firmware-nonfree-rpi-distro-bcm43455-0.0+git0+b518de45ce-r0.1.noarch.rpm
firmware-nonfree-rpi-distro-broadcom-license-0.0+git0+b518de45ce-r0.1.noarch.rpm
```
to become:
```
linux-firmware-rpidistro-bcm43430a1-hcd-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
linux-firmware-rpidistro-bcm4345c0-hcd-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
linux-firmware-rpidistro-cypress-license-0.0+git0+ade2bae1aa-r0.2.noarch.rpm
linux-firmware-rpidistro-bcm43430-0.0+git0+b518de45ce-r0.1.noarch.rpm
linux-firmware-rpidistro-bcm43455-0.0+git0+b518de45ce-r0.1.noarch.rpm
linux-firmware-rpidistro-broadcom-license-0.0+git0+b518de45ce-r0.1.noarch.rpm
```
even though none of the content comes from the maintained linux-firmware distribution and the package names will not match the recipe names.  Note that I can't produce all six from one recipe: the material from bluez-firmware and firmware-nonfree must remain separate recipes.  If you'd accept leaving the first three as `bluez-firmware-rpidistro-` instead of `linux-firmware-rpidistro-` I can make the recipe names consistent with the packages, which would probably be a good thing.